tests/test_contacts.cpp

```diff
@@ -41,42 +41,810 @@ namespace tests
         }
     }
 
+    TEST(OfflineContactTest, InitialFileAs)
+    {
+        auto minnie = ews::contact();
+        EXPECT_STREQ("", minnie.get_file_as().c_str());
+    }
+
+    TEST(OfflineContactTest, SetFileAs)
+    {
+        auto minnie = ews::contact();
+        minnie.set_file_as("Minnie Mouse");
+        EXPECT_STREQ("Minnie Mouse", minnie.get_file_as().c_str());
+    }
+
+    TEST_F(ContactTest, UpdateFileAs)
+    {
+        auto minnie = test_contact();
+        auto prop =
+            ews::property(ews::contact_property_path::file_as, "Minnie Mouse");
+        auto new_id = service().update_item(minnie.get_item_id(), prop);
+        minnie = service().get_contact(new_id);
+        EXPECT_STREQ("Minnie Mouse", minnie.get_file_as().c_str());
+    }
+
+    TEST(OfflineContactTest, InitialFileAsMappingValue)
+    {
+        auto minnie = ews::contact();
+        EXPECT_STREQ("", minnie.get_file_as_mapping().c_str());
+    }
+
+    TEST(OfflineContactTest, SetFileAsMappingValue)
+    {
+        auto minnie = ews::contact();
+        minnie.set_file_as_mapping("LastCommaFirst");
+        EXPECT_STREQ("LastCommaFirst", minnie.get_file_as_mapping().c_str());
+    }
+
+    TEST_F(ContactTest, UpdateFileAsMappingValue)
+    {
+        auto minnie = test_contact();
+        auto prop =
+            ews::property(ews::contact_property_path::file_as, "Minnie Mouse");
+        auto new_id = service().update_item(minnie.get_item_id(), prop);
+        minnie = service().get_contact(new_id);
+        EXPECT_STREQ("Minnie Mouse", minnie.get_file_as().c_str());
+
+        prop = ews::property(ews::contact_property_path::file_as_mapping,
+                             "LastSpaceFirst");
+        new_id = service().update_item(minnie.get_item_id(), prop);
+        minnie = service().get_contact(new_id);
+        EXPECT_STREQ("LastSpaceFirst", minnie.get_file_as_mapping().c_str());
+        EXPECT_STREQ("Mouse Minerva", minnie.get_file_as().c_str());
+    }
+
+    TEST(OfflineContactTest, InitialEmailAddressProperty)
+    {
+        auto minnie = ews::contact();
+        EXPECT_TRUE(minnie.get_email_addresses().empty());
+    }
+
+    TEST(OfflineContactTest, SetEmailAddressProperty)
+    {
+        auto minnie = ews::contact();
+        auto email = ews::email_address(ews::email_address::key::email_address_1,
+                    "minnie.mouse@duckburg.com");
+        minnie.set_email_address(email);
+        auto new_mail =  minnie.get_email_addresses();
+        EXPECT_EQ(email, new_mail[0]);
+    }
```

Comment:
Make sure you assert first the expected size of a vector before accessing one of its elements. This makes failing test-cases really fail and not segfault.

```c++
    TEST(OfflineContactTest, SetEmailAddressProperty)
    {
        auto minnie = ews::contact();
        auto email = ews::email_address(ews::email_address::key::email_address_1,
                    "minnie.mouse@duckburg.com");
        minnie.set_email_address(email);
        auto addresses = minnie.get_email_addresses();
        ASSERT_EQ(1U, addresses.size());        // <----
        EXPECT_EQ(email, addresses[0]);
    }
```

Other tests accordingly.